lib/swagger-module.ts

```diff
@@ -35,87 +45,107 @@ export class SwaggerModule {
     };
   }
 
+  private static serveStatic(finalPath: string, app: INestApplication) {
+    const httpAdapter = app.getHttpAdapter();
+
+    if (httpAdapter && httpAdapter.getType() === 'fastify') {
+      (app as NestFastifyApplication).useStaticAssets({
+        root: swaggerAssetsAbsoluteFSPath,
+        prefix: `${finalPath}/`,
+        decorateReply: false
+      });
+    } else {
+      (app as NestExpressApplication).useStaticAssets(
+        swaggerAssetsAbsoluteFSPath,
+        { prefix: finalPath }
+      );
+    }
+  }
+
+  private static serveDocuments(
+    finalPath: string,
+    httpAdapter: HttpServer,
+    swaggerInitJS: string,
+    yamlDocument: string,
+    jsonDocument: string,
+    html: string
+  ) {
+    httpAdapter.get(`${finalPath}/swagger-ui-init.js`, (req, res) => {
+      res.type('application/javascript');
+      res.send(swaggerInitJS);
+    });
+
+    httpAdapter.get(finalPath, (req, res) => {
+      res.type('text/html');
+      res.send(html);
+    });
+
+    // fastify compatibility
+    httpAdapter.get(finalPath + '/', (req, res) => {
+      res.type('text/html');
+      res.send(html);
+    });
+
+    httpAdapter.get(`${finalPath}-json`, (req, res) => {
+      res.type('text/json');
+      res.send(jsonDocument);
+    });
+
+    httpAdapter.get(`${finalPath}-yaml`, (req, res) => {
+      res.type('text/yaml');
+      res.send(yamlDocument);
+    });
+  }
+
   public static setup(
     path: string,
     app: INestApplication,
     document: OpenAPIObject,
-    options?: SwaggerCustomOptions
+    options?: SwaggerCustomOptionsLegacy
   ) {
-    const httpAdapter = app.getHttpAdapter();
     const globalPrefix = getGlobalPrefix(app);
     const finalPath = validatePath(
       options?.useGlobalPrefix && globalPrefix && !globalPrefix.match(/^(\/?)$/)
         ? `${globalPrefix}${validatePath(path)}`
         : path
     );
-    if (httpAdapter && httpAdapter.getType() === 'fastify') {
-      return this.setupFastify(
-        finalPath,
-        httpAdapter,
-        document,
-        options as FastifySwaggerCustomOptions
-      );
-    }
-    return this.setupExpress(
-      finalPath,
-      app,
-      document,
-      options as ExpressSwaggerCustomOptions
-    );
-  }
 
-  private static setupExpress(
-    path: string,
-    app: INestApplication,
-    document: OpenAPIObject,
-    options?: ExpressSwaggerCustomOptions
-  ) {
+    // START: fastify backward compatibility layer
+    const { customOptions, extra } = processSwaggerOptions(options);
+    // END: fastify backward compatibility layer
```

Comment:
Just to make sure that I'm following - once we decide to remove the backward compatibility layer, we can simply remove these lines? And then `options` === `customOptions`? What about `extra`?

Reply:
extra is the diff that is currently supported only in fastify (uiHooks).
Let's clarify the end goal (thread under [lib/backward-compatilibity-layer.ts](https://github.com/nestjs/swagger/pull/1886/files/92e85c9c2d2c0f979cd9d4244d492bdaae706b6f#diff-45b88e46065760c5d355b77cd01bb5155c53012ffcc7a5131275639099de34fe))
and I would be able to answer this

Reply:
1. set setup method param 'options' type to SwaggerCustomOptions
2. drop usage of processSwaggerOptions (use customOptions = options)
3. drop usage of serveDocumentsFastify
4. remove backward-compatility-layer.ts file

and we need to gradually update documentation on https://docs.nestjs.com/openapi/introduction